ocamltest/ocaml_actions.ml

```diff
@@ -653,31 +616,28 @@ let finalise_codegen_msvc test_basename log env =
   let what = "Running Microsoft assembler" in
   Printf.fprintf log "%s\n%!" what;
   let commandline = [Ocamltest_config.asm; obj; src] in
-  let expected_exit_status = 0 in
-  let exit_status =
+  let result =
     Actions_helpers.run_cmd
       ~environment:default_ocaml_env
       ~stdout_variable:Ocaml_variables.compiler_output
       ~stderr_variable:Ocaml_variables.compiler_output
       ~append:true
+      ~what
       log env commandline in
-  if exit_status=expected_exit_status
-  then begin
-    let archmod = Ocaml_files.asmgen_archmod in
-    let modules = obj ^ " " ^ archmod in
-    let program = Filename.make_filename test_basename "out" in
-    let env = Environments.add_bindings
-    [
-      Ocaml_variables.modules, modules;
-      Builtin_variables.program, program;
-    ] env in
-    (Result.pass, env)
-  end else begin
-    let reason =
-      (Actions_helpers.mkreason
-        what (String.concat " " commandline) exit_status) in
-    (Result.fail_with_reason reason, env)
-  end
+  let env =
+    if not (Result.is_pass result) then
+      env
+    else
```

Comment:
I would always use `begin`..`end` here, to avoid later issues if someone adds a sequence in the expression.

Reply:
Note: Another way to write this code would be to move the `bindings` declaration outside the conditional (as a unit function to delay effects), so that both branches are very short.

Reply:
Note: seeing the code below, this logic could in fact move to a `finalise_*` function -- in an independent commit.